src/api/client_server/sync.rs

```diff
@@ -705,7 +705,11 @@ async fn load_joined_room(
 				// Recalculate heroes (first 5 members)
 				let mut heroes = Vec::new();
 
-				if joined_member_count + invited_member_count <= 5 {
+				if joined_member_count
+					.checked_add(invited_member_count)
+					.expect("joined/invite member count should not be this high")
```

Comment:
You're only trying to check if it's `<= 5`, but panicking if it's really large? Use `.saturating_add()`.